feat: add `roundOffsets` option to computeStyles modifier

piecyk opened the pull request:
As reported in #1169 rounding logic in some cases makes the popper not correctly aligned with reference element. This PR adds possibility to opt out from it via roundOffsets option.

```js
createPopper(reference, popper, {
  modifiers: [
    {
      name: 'computeStyles',
      options: {
        roundOffsets: false,
      },
    },
  ],
});
```

FezVrasta commented:
Can't we fix the rounding algorithm instead? We have quite a lot of tests that cover it already.

piecyk commented:
> Can't we fix the rounding algorithm instead? We have quite a lot of tests that cover it already.

I think it will be hard to make a fix that will work cross browsers.

Also while testing locally latest chrome/firefox/edge/safari with opting out, didn't break the rendering as mainly that's the reason for rounding.

piecyk commented on src/utils/computeOffsets.js:
```diff
@@ -68,17 +84,15 @@ export default function computeOffsets({
     switch (variation) {
       case start:
         offsets[mainAxis] =
-          Math.floor(offsets[mainAxis]) -
```
Why we had this rounds functions on start/end variation? Imho they are not needed as  latter we still round the offsets 🤔

FezVrasta replied:
@atomiks? 👀

piecyk replied:
Maybe I should add more test 🤔 Also this change didn't break any existing one.

FezVrasta replied:
The PR looks good, I'm just a bit worried about this bit. Let's give @atomiks a couple more days to review it and if he doesn't find time I'll merge it as dog food it to the whole user base :-)

piecyk replied:
Looks like no other comments. IMHO let's merge this 🤞 💪

piecyk commented on src/utils/computeOffsets.js:
```diff
@@ -7,23 +7,39 @@ import type {
+    x: Math.round(x * dpr) / dpr || 0,
+    y: Math.round(y * dpr) / dpr || 0,
+  };
+}
```
I think there is no way to correctly round the offsets using `devicePixelRatio` Maybe we can just go with something simpler, ping @atomiks 

```js
// Round the offsets to the nearest subpixel.
function roundOffsets({ x = 0, y = 0 }): Offsets {
  return {
    x: Math.round(x * 10) / 10 || 0,
    y: Math.round(y * 10) / 10 || 0,
  };
}
```

piecyk replied:
Let's keep the rounding logic in computeStyles.

piecyk commented:
Don't really see a way to improve rounding algorithm to be more accurate 🤷‍♂️ Maybe till then we can have this opt out option.

piecyk commented:
Changed this to keep the rounding logic in computeStyles modifier, added `roundOffsets` option as boolean to modifier options as in PR description. Was wondering also if not allow to pass custom function, so then user of modifier can tweak the rounding.

FezVrasta commented:
Merged, thanks for the help! Could you also send a PR on the `computeStyles` docs? [here](https://github.com/popperjs/website/edit/master/src/pages/docs/v2/modifiers/compute-styles.mdx)

piecyk commented:
> Could you also send a PR on the computeStyles docs? 

Yes, will do.

piecyk commented:
@FezVrasta when are you planing releasing this? I think patch release will be good in this case. Also thanks for this 👏

FezVrasta commented:
Releasing
